.github/workflows/build.yml

```diff
@@ -53,9 +56,11 @@ jobs:
           global-json-file: ${{ matrix.global-json-file }}
           dotnet-version: ${{ matrix.dotnet-version }}
 
-      # remove global.json so that the env configuration takes precedence
+      # remove global.json, create new one to protect against CI machine installed defaults. Then the env configuration takes precedence
       - name: Purge global.json
-        run: rm global.json
+        run: |
+          rm global.json
+          ${{ matrix.globaljson-command }}
```

Comment:
As mentioned in the comment, the CI machines we run on still have other dotnet versions installed on them. This helps protect against that.